packages/fxa-payments-server/src/routes/Subscriptions/ActionButton.tsx

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { Localized } from '@fluent/react';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+import { Customer } from '../../store/types';
+
+import * as PaymentProvider from '../../lib/PaymentProvider';
+
+export type ActionButtonProps = {
+  customer: Customer;
+  onRevealUpdateClick: () => void;
+  revealFixPaymentModal: () => void;
+};
+
+export const ActionButton = ({
+  customer,
+  onRevealUpdateClick,
+  revealFixPaymentModal,
+}: ActionButtonProps) => {
+  const {
+    billing_agreement_id,
+    payment_provider,
+    paypal_payment_error,
+  } = customer;
+
+  const stripeActionButton = () => (
+    <button
+      data-testid="reveal-payment-update-button"
+      className="settings-button"
+      onClick={onRevealUpdateClick}
+    >
+      <Localized id="pay-update-change-btn">
+        <span className="change-button" data-testid="change-button">
+          Change
+        </span>
+      </Localized>
+    </button>
+  );
+
+  const paypalActionButton = () => (
+    <Localized id="pay-update-change-btn">
+      <LinkExternal
+        data-testid="change-payment-update-button"
+        className="settings-button"
+        href={`https://www.paypal.com/myaccount/autopay/connect/${billing_agreement_id}`}
```

Comment:
The server side changes (the optional property on the type and the response validator) are saying that `billing_agreement_id` could be undefined, but in this component we use it without handling that state.

Reply:
My understanding is that if `billing_agreement_id` is missing, then `paypal_payment_error` should be `missing_agreement` in which we will return `paypalMissingAgreementActionButton`.

Is there a case where `billing_agreement_id` would come back undefined without `paypal_payment_error` being set?

Reply:
Ah, I see.  Thanks for the info!